packages/rest-specs/specs/authentication/api-key/mockapi.ts

```diff
@@ -0,0 +1,58 @@
+import { MockRequest, passOnSuccess, ScenarioMockApi } from "@typespec/spec-api";
+import { getValidAndInvalidScenarios } from "../commonapi.js";
+
+export const Scenarios: Record<string, ScenarioMockApi> = {};
+
+const validAndInvalidScenarios = getValidAndInvalidScenarios(
+  "api-key",
+  "invalid-api-key",
+  function addOptionalParamOldApiVersionNewClientValidate(req: MockRequest): void {
+    req.expect.containsHeader("x-ms-api-key", "valid-key");
+  },
+);
+
+Scenarios.Authentication_ApiKey_valid = validAndInvalidScenarios.valid;
+
+Scenarios.Authentication_ApiKey_invalid = validAndInvalidScenarios.invalid;
+
+Scenarios.Authentication_ApiKey_InValid_Server_Test = passOnSuccess({
```

Comment:
would hope that in those case where the client was using some reusable function to build the mock api  we'd do the same for the definition.

Reply:
Yes. In most scenarios, they are generated using the resusable functions